Sources/Datadog/Datadog.swift

```diff
@@ -166,49 +224,113 @@ public class Datadog {
+        )
 
         // Set default `DatadogCore`:
         let core = DatadogCore(
-            directory: try CoreDirectory(in: Directory.cache(), from: configuration.common),
-            dateProvider: configuration.common.dateProvider,
-            initialConsent: initialTrackingConsent,
-            performance: configuration.common.performance,
-            httpClient: HTTPClient(proxyConfiguration: configuration.common.proxyConfiguration),
-            encryption: configuration.common.encryption,
+            directory: try CoreDirectory(
+                in: Directory.cache(),
+                clientToken: configuration.clientToken,
+                site: configuration.site
+            ),
+            dateProvider: configuration.dateProvider,
+            initialConsent: trackingConsent,
+            performance: performance,
+            httpClient: HTTPClient(proxyConfiguration: configuration.proxyConfiguration),
+            encryption: configuration.encryption,
             contextProvider: DatadogContextProvider(
-                configuration: configuration.common,
-                device: .init(),
-                serverDateProvider: serverDateProvider
+                site: configuration.site,
+                clientToken: try ifValid(clientToken: configuration.clientToken),
+                service: service,
+                env: try ifValid(env: configuration.env),
+                version: applicationVersion,
+                variant: variant,
+                source: source,
+                sdkVersion: sdkVersion,
+                ciAppOrigin: CITestIntegration.active?.origin,
+                applicationName: bundleName ?? bundleType.rawValue,
+                applicationBundleIdentifier: bundleIdentifier,
+                applicationVersion: applicationVersion,
+                sdkInitDate: configuration.dateProvider.now,
+                device: DeviceInfo(),
+                dateProvider: configuration.dateProvider,
+                serverDateProvider: configuration.serverDateProvider
             ),
-            applicationVersion: configuration.common.applicationVersion
+            applicationVersion: applicationVersion
         )
 
         let telemetry = TelemetryCore(core: core)
 
         telemetry.configuration(
-            batchSize: Int64(exactly: configuration.common.performance.maxFileSize),
-            batchUploadFrequency: configuration.common.performance.minUploadDelay.toInt64Milliseconds,
-            useLocalEncryption: configuration.common.encryption != nil,
-            useProxy: configuration.common.proxyConfiguration != nil
+            batchSize: Int64(exactly: performance.maxFileSize),
+            batchUploadFrequency: performance.minUploadDelay.toInt64Milliseconds,
+            useLocalEncryption: configuration.encryption != nil,
+            useProxy: configuration.proxyConfiguration != nil
         )
 
-        CITestIntegration.active?.startIntegration()
-
```

Comment:
question/ Where are we moving this to? Unfortunately, `CITestIntegration` is covered with integration tests and it is easy to mess it.

Reply:
Good catch, it was a miss 🙏